### PR TITLE
Accounting SWAP protocol basic data types and test

### PR DESCRIPTION
**Pull request by oskarth**
Partial progress towards https://github.com/status-im/nim-waku/issues/260

Ready to be reviewed and merged.

Making it more end to end and integrated will be done in subsequent PRs.

### CODE REVIEW

File: waku/node/v2/waku_types.nim [hunk 1]
```diff
@@ -146,8 +148,21 @@ type
...
+  Beneficiary* = seq[byte]
+
+  # TODO Consider adding payment threshhold and terms field
+  Handshake* = object
+    beneficiary*: Beneficiary
+
+  Cheque* = object
+    beneficiary*: Beneficiary
+    date*: uint32
+    amount*: uint32
```
Comment (staheri14) [inline waku/node/v2/waku_types.nim:161]: For the sake of authentication, a signature field can be added to the Cheque.

File: tests/v2/test_waku_swap.nim [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    check:
+      decodedHandshake.isErr == false
+      decodedHandshake.get().beneficiary == beneficiary
+
+  test "Cheque Encode/Decode":
+    let
+      amount = 1'u32
+      date = 9000'u32
+      beneficiary = @[byte 0, 1, 2]
+      cheque = Cheque(beneficiary: beneficiary, amount: amount, date: date)
```
Comment (decanus) [inline tests/v2/test_waku_swap.nim:26]: This is probably an issue for the specs but I am not sure i like the naming of Cheque

File: waku/node/v2/waku_types.nim (hunk 1 above)
Comment (oskarth) [reply waku/node/v2/waku_types.nim:161]: Yep, I'll do that in follow up PRs. Still thinking about how this should work exactly

File: tests/v2/test_waku_swap.nim (hunk 2 above)
Comment (oskarth) [reply tests/v2/test_waku_swap.nim:26]: Why? It describes what it does and it is what is used in the Swarm book. We could change it, but we'd have to have a good rationale and alternative then.

**Comment by oskarth** [comment]
@jm-clius @decanus @staheri14 From my point of view this can be merged, let me know if there are any concerns. Changes are quite isolated so should have minimal impact on other parts of the code base. Further capabilities to be introduced in follow up PRs.

**Pull request merged**